templates/nginx.conf.j2

```diff
@@ -80,3 +81,12 @@ server {
         deny all;
     }
 }
+
+{% if laravel_https_redirect -%}
+server {
+    listen         80;
+    listen         [::]:80;
+    server_name    {{ server_name }};
+    return         301 https://{{ server_name }}$request_uri;
```

Comment:
Use `$host` instead of `{{ server_name }}`. The server_name variable can be a regex, while host is the one requested by browser.